action_server/src/action_server/server.py

```diff
@@ -36,7 +37,15 @@ def _get_actions_cb(self, req):
     def _add_action_cb(self, goal):
         recipe = yaml.load(goal.recipe)
         rospy.logdebug("Received action recipe: {}".format(goal.recipe))
-        configuration_result = self._task_manager.set_up_state_machine(recipe['actions'])
+
+        # check if we have a recipe that makes sense, otherwise set result to failed
+        # note that python uses lazy evaluation of boolean conditions
+        if isinstance(recipe, dict) and 'actions' in recipe and recipe['actions']:
+            configuration_result = self._task_manager.set_up_state_machine(recipe['actions'])
+        else:
+            rospy.logerr('Recipe does not contain actions, setting configuration result to failed')
+            configuration_result = ConfigurationResult()
```

Comment:
You could add a message: `configuration_result.message = 'bla'`. This will be passed to the client, which can report back to the user.